src/components/DonationCardModal/DonationCardModal.tsx

```diff
@@ -0,0 +1,86 @@
+import { useState } from 'react';
+import styles from './DonationCardModal.module.css';
+
+interface DonationCardModalProps {
+   title: string;
+   target: number;
+   amount: number;
+   onUpdateDonationAmount: (amount: number) => void;
+   onCloseModal: () => void;
+}
+
+function DonationCardModal({
+   title,
+   target,
+   amount,
+   onUpdateDonationAmount,
+   onCloseModal,
+}: DonationCardModalProps) {
+   const [donationAmount, setDonationAmount] = useState(0);
+
+   const handleBackgroundClick = (event: React.MouseEvent<HTMLDivElement>) => {
+      if (event.target === event.currentTarget) {
+         onCloseModal();
+      }
+   };
+
+   const handleAmountChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+      setDonationAmount(Number(event.target.value));
+   };
+
+   const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+      onUpdateDonationAmount(donationAmount);
+      onCloseModal();
+   };
+
+   return (
+      <div className={styles.modal} onClick={handleBackgroundClick}>
+         <form className={styles.modalContent} onSubmit={handleFormSubmit}>
+            <header className={`${styles.modalBoundary} ${styles.modalHeader}`}>
+               <div className={styles.modalHeaderContent}>
+                  <h2>{title}</h2>
+                  <button
+                     type="button"
+                     className={styles.xButton}
+                     onClick={onCloseModal}>
+                     &times;
+                  </button>
+               </div>
+            </header>
```

Comment:
Couldn't the ```<header>``` be enough without the use of the ```<div>``` too? why 2 wrapping elements? 
Can you try to remove the ```<div>``` and play with the styling with the rest of the elements?